src/rcx/src/netRC.cpp

```diff
@@ -1881,9 +1874,11 @@ bool extMain::setCorners(const char* rulesFileName, const char* cmp_file)
     logger_->info(
         RCX, 435, "Reading extraction model file {} ...", rulesFileName);
 
-    if (!fopen(rulesFileName, "r"))
+    FILE* rules_file = fopen(rulesFileName, "r");
+    if (rules_file == nullptr)
       logger_->error(
           RCX, 468, "Can't open extraction model file {}", rulesFileName);
+    fclose(rules_file);
```

Comment:
error() will throw an exception and the file will be left open.  Probably an iostream would be simpler but otherwise you need to fclose before the error

Reply:
It's good to know that exceptions are enabled and the error routine throws, thanks for pointing that out.

I think this is ok in that circumstance though -- if the file handle is `nullptr` there's nothing to close, right? We only need to close in the case we successfully opened the file descriptor. (Really we should be using like `std::filesystem::exists` or a `stat` variant but I wanted to try to make minimal changes in this CL just to get my bearings :-)

Reply:
fair point